### PR TITLE
Fixes issue #1011

### LINKED ISSUE #1009: problems for geometric selection command 'prop'
I used the keyword 'prop' to select atoms within a range of absolute coordinate.
But it showed me error. I have no idea about this error, I just typed the command as the online documentation says. Could someone help me ?
### Code to I used:

import MDAnalysis as md
u=md.Universe('OW_2us.pdb')
sel=u.select_atoms("prop 1<=x")
### Error message:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python2.7/dist-packages/MDAnalysis/core/AtomGroup.py", line 4922, in select_atoms
    return self.atoms.select_atoms(sel, *othersel, **selgroups)
  File "/usr/local/lib/python2.7/dist-packages/MDAnalysis/core/AtomGroup.py", line 3310, in select_atoms
    atomgrp = Selection.Parser.parse(selstr, selgroups).apply(self)
  File "/usr/local/lib/python2.7/dist-packages/MDAnalysis/core/Selection.py", line 1000, in parse
    parsetree = self.parse_expression(0)
  File "/usr/local/lib/python2.7/dist-packages/MDAnalysis/core/Selection.py", line 1008, in parse_expression
    exp1 = self._parse_subexp()
  File "/usr/local/lib/python2.7/dist-packages/MDAnalysis/core/Selection.py", line 1026, in _parse_subexp
    return _SELECTIONDICT[op](self, self.tokens)
  File "/usr/local/lib/python2.7/dist-packages/MDAnalysis/core/Selection.py", line 839, in __init__
    self.value = float(tokens.popleft())
IndexError: pop from an empty deque
```
### currently version of mdanalysis:

0.15.0

....

```


```

### LINKED ISSUE #1011: Selection parsing not splitting around comparison operators nicely
### Current behaviour

Currently doing prop selections requires whitespace around the comparison operators, 

``` python
import MDAnalysis as mda
from MDAnalysisTests.datafiles import PSF, DCD

u = mda.Universe(PSF, DCD)

u.select_atoms('prop x <= 5')  # this works!

u.select_atoms('prop x <=5')  # this fails!

u.select_atoms('prop x<=5')  # this also fails!
```
### Desired behaviour

The following strings should all work and return the same result:

``` python
u.select_atoms('prop x <= 5')  # working currently!
u.select_atoms('prop x <=5')  # no space between <= and 5
u.select_atoms('prop x<= 5')  # no space between x and <=
u.select_atoms('prop x<=5')  # no spaces anywhere 

# Second set of tests
u.select_atoms('prop x > 10')
# Should give the same as
u.select_atoms('prop x >10')
u.select_atoms('prop x> 10')
u.select_atoms('prop x>10')
```

This should be possible through adding extra splitting such as [what happens for brackets](https://github.com/MDAnalysis/mdanalysis/blob/develop/package/MDAnalysis/core/Selection.py#L1004).

Alternatively, it could be handled inside the [PropSelection](https://github.com/MDAnalysis/mdanalysis/blob/develop/package/MDAnalysis/core/Selection.py#L837) object itself, by checking if the token it is looking at requires further splitting.  This would allow the corner case of atom names which contain comparison operator characters. (Eg `C<` if anyone would do this?)

Some care is required so that comparison operators with 2 characters (`==` `<=` etc) aren't split twice by naively splitting around all comparison operators.

Originally raised as part of #1009

### PR DESCRIPTION
**Pull request by richardjgowers**
Fixes #1011 
Fixes #1009

Changes made in this Pull Request:
 - property selection strings are now less strict on having spaces (see #1009)

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

File: package/MDAnalysis/core/selection.py [hunk 1]
```diff
@@ -963,14 +963,58 @@ class PropertySelection(Selection):
...
+            try:
+                x, y = oper.split(possible)
+                if y:
+                    oper = possible
+                    value = y
+            except ValueError:
+                pass
+        else:  # if none of the previous triggers
+            if not ('<=' in oper or '>=' in oper):  # to stop splitting around these
+                for possible in ('<', '>'):
```
Comment (jbarnoud) [inline package/MDAnalysis/core/selection.py]: Why are these cases handled separately?

**Comment by jbarnoud** [comment]
I've been to slow to review. You fixed all my comments already :)

File: package/MDAnalysis/core/selection.py (hunk 1 above)
Comment (richardjgowers) [inline package/MDAnalysis/core/selection.py]: It was because `'<'` would split around `'<='` causing `'='` to be the "value", I've refactored

**Comment by richardjgowers** [comment]
@jbarnoud replaced by machines!

File: package/MDAnalysis/core/selection.py [hunk 2]
```diff
@@ -963,14 +963,47 @@ class PropertySelection(Selection):
...
+                pass
+            else:
+                prop = x
+                oper = possible + y  # add back after splitting
+                break
+
+        if oper is None:
+            oper = tokens.popleft()
+        # check if oper has the value appended
+        for possible in ('<=', '>=', '==', '!=', '<', '>'):
```
Comment (jbarnoud) [inline package/MDAnalysis/core/selection.py]: The tuple of operators could go in a separate variable to avoid being repeated and ease the addition of new operators.

File: package/MDAnalysis/core/selection.py [hunk 3]
```diff
@@ -979,6 +1012,7 @@ def __init__(self, parser, tokens):
             raise ValueError(
                 "Invalid operator : '{0}' Use one of : '{1}'"
                 "".format(oper, self.ops.keys()))
+        self.value = float(value)
```
Comment (jbarnoud) [inline package/MDAnalysis/core/selection.py]: Wouldn't something like:

```python
opposites = {'<': '>', '>': '<', '<=': '>=', '>=': '<='}
try:
    self.value = float(value)
except ValueError:
    try:
        self.value = float(prop)
    except ValueError as e:
        raise e
    else:
        self.prop = value
        self.oper = opposites.get(oper, oper)
```

allow to have both `x > 5` and `5 < x`?
allow to write

Comment (richardjgowers) [inline package/MDAnalysis/core/selection.py]: Yep I did think along those lines.  I was also thinking if there'd be a neat way of implementing chained comparisons (`10 < x < 15`) by peeking at the next token and seeing if it's a operator.

So

```python
if tokens[0] in self.ops:
    # split this into two selections, checking if it logically makes sense...
    return AndOperation(firstsel, secondsel)
```

Comment (jbarnoud) [inline package/MDAnalysis/core/selection.py]: It would be nice as people working in python expect chained comparison to work. It seems slightly out of scope for this PR so, beside my [small comment above](https://github.com/MDAnalysis/mdanalysis/pull/1156#discussion_r95631318), the PR looks good to go. Would you like to have a look at inverted syntax and chained comparisons or shall we open new issues?

Comment (richardjgowers) [inline package/MDAnalysis/core/selection.py]: I'm months ahead of you #1012 :P

Comment (richardjgowers) [inline package/MDAnalysis/core/selection.py]: But yeah, we'll do it in another PR, I think getting the AndOperation working right will be a small headache

**Comment by jbarnoud** [comment]
I was about to merge when I realize you did not update the CHANGELOG.
